catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/formulaeditor/NewDataDialogTest.java

```diff
@@ -0,0 +1,222 @@
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.uiespresso.formulaeditor;
+
+import android.support.test.runner.AndroidJUnit4;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.content.Script;
+import org.catrobat.catroid.content.bricks.AddItemToUserListBrick;
+import org.catrobat.catroid.content.bricks.SetVariableBrick;
+import org.catrobat.catroid.ui.SpriteActivity;
+import org.catrobat.catroid.uiespresso.content.brick.utils.BrickTestUtils;
+import org.catrobat.catroid.uiespresso.testsuites.Cat;
+import org.catrobat.catroid.uiespresso.testsuites.Level;
+import org.catrobat.catroid.uiespresso.util.rules.BaseActivityInstrumentationRule;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.action.ViewActions.typeText;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickDataInteractionWrapper.onBrickAtPosition;
+
+@RunWith(AndroidJUnit4.class)
+public class NewDataDialogTest {
+
+	@Rule
+	public BaseActivityInstrumentationRule<SpriteActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(SpriteActivity.class, SpriteActivity.EXTRA_FRAGMENT_POSITION, SpriteActivity.FRAGMENT_SCRIPTS);
+
+	private int variableBrickPosition;
+	private int listBrickPosition;
+
+	@Before
+	public void setUp() throws Exception {
+		Script startScript = BrickTestUtils.createProjectAndGetStartScript("newDataDialogTest");
+
+		variableBrickPosition = 1;
+		listBrickPosition = 2;
+
+		// "when program starts" brick is the first
+		startScript.addBrick(variableBrickPosition - 1, new SetVariableBrick());
+		startScript.addBrick(listBrickPosition - 1, new AddItemToUserListBrick());
+		baseActivityTestRule.launchActivity();
+	}
+
+	@Category({Cat.AppUi.class, Level.Smoke.class})
+	@Test
+	public void testCreateNewVariable() {
+		String normalVarName = "myVar";
+
+		onBrickAtPosition(variableBrickPosition).onVariableSpinner(R.id.set_variable_spinner)
+				.performNewVariableInitial(normalVarName);
+	}
+
+	@Category({Cat.AppUi.class, Level.Functional.class})
+	@Test
+	public void testCorrectDialogTitle() {
+		openNewDataDialog(variableBrickPosition, R.id.set_variable_spinner);
+
+		onView(withText(R.string.formula_editor_variable_dialog_title))
+				.noActivity();
```

Comment:
why the noActivity?
this whole line doesnt do anything, and doesnt assert anything. it doesnt even check if the dialog title is displayed. if there is no view fitting your matcher and you do not perform anything on it, nothing is done.
you'd have to add .check(isDisplayed()) to assert that it is there.